### PR TITLE
kv: add a TODO warning of broken txn init

### PR DESCRIPTION
**Pull request by andreimatei**

### CODE REVIEW

File: pkg/kv/dist_sender.go [hunk 1]
```diff
@@ -516,6 +516,10 @@ func (ds *DistSender) initAndVerifyBatch(
 		if len(ba.Txn.ObservedTimestamps) == 0 {
 			// Ensure the local NodeID is marked as free from clock offset;
 			// the transaction's timestamp was taken off the local clock.
+			// TODO(andrei): This is broken when Txn.OrigTimestamp has not been, in
```
Comment (nvanbenschoten) [inline pkg/kv/dist_sender.go]: Could you explain why it is broken? What can happen in the case where the `txn.OrigTimestamp` was set from another node's clock?

**Review by bdarnell — APPROVED** [review]
LGTM, although an issue may be a better place for this than a TODO.

**Comment by andreimatei** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/kv/dist_sender.go, line 519 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14730#-KhJUAnQj3H0_0r1INbZ-r1-519:-KhOFE6DUM9HM3PoTE0c:be5ntg6) ([raw file](https://github.com/cockroachdb/cockroach/blob/aaa07f1c8e0c7aeb16fa6ed8076b85f6d2a07da0/pkg/kv/dist_sender.go#L519)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Could you explain why it is broken? What can happen in the case where the `txn.OrigTimestamp` was set from another node's clock?
</blockquote></details>

This TODO is in direct relation to the statement above; I don't think it needs more explanation...

What would happen is that the reasoning for this code goes out the window and we're pretending we've observed a timestamp that in fact we haven't observed. The consequence is that values encountered on this gateway node with timestamps greater than the OrigTimestamp will not be considered ambiguous, although they should be.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14730)*

**Comment by andreimatei** [comment]
Ben, we already have an issue mentioning that `ExternalClient` should go. I don't think another open issue about this specific thing would be useful...

**Pull request merged**